crates/ruff_linter/src/rules/flake8_pyi/rules/string_or_bytes_too_long.rs

```diff
@@ -100,6 +101,34 @@ fn count_f_string_chars(f_string: &ast::ExprFString) -> usize {
         .sum()
 }
 
+/// Count the number of visible characters in an f-string. This accounts for
+/// implicitly concatenated f-strings as well.
+fn count_t_string_chars(f_string: &ast::ExprTString) -> usize {
+    f_string
+        .value
+        .iter()
+        .map(|part| match part {
+            ast::TStringPart::Literal(string) => string.chars().count(),
+            ast::TStringPart::FString(f_string) => f_string
+                .elements
+                .iter()
+                .map(|element| match element {
+                    ast::FStringElement::Literal(string) => string.chars().count(),
+                    ast::FStringElement::Expression(expr) => expr.range().len().to_usize(),
+                })
+                .sum(),
```

Comment:
Consider adding a `count_fstring_chars` (and rename the existing one to `_expr`) to share the f-string logic instead of repeating it here